src/engine/macros.rs

```diff
@@ -23,10 +23,14 @@ macro_rules! destroy_filesystems {
     ( $s:ident; $fs:expr ) => {
         let mut removed = Vec::new();
         for uuid in $fs.iter().map(|x| *x) {
+            match $s.filesystems.get_mut_by_uuid(&uuid) {
```

Comment:
I _think_ this match case is redundant. Probably you can call ```remove()``` on the thing returned by ```remove_by_uuid()```